### PR TITLE
Add EndpointConfig.properties templating.

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed Changes**

- Add EndpointConfig.properties templating.
- Read property values only if tenantListEnabled=true or mutualSSLManagerEnabled=true.
- Set different keystore and userstore filename and password.

### CODE REVIEW

File: modules/distribution/src/repository/resources/conf/infer.json [hunk 1]
```diff
@@ -115,5 +115,37 @@
...
+
+  "identity.auth_framework.endpoint.mutual_ssl_manager_enabled": {
+    "true": {
+      "identity.auth_framework.endpoint.hostname_verification_enabled" : true,
+      "identity.auth_framework.endpoint.mutual_ssl_username" : "admin",
+      "identity.auth_framework.endpoint.client_keystore" : "$conf{keystore.tls.file_name}",
+      "identity.auth_framework.endpoint.carbon_security_keystore_password" : "$conf{keystore.tls.password}",
+      "identity.auth_framework.endpoint.client_truststore" : "$conf{truststore.file_name}",
+      "identity.auth_framework.endpoint.carbon_security_truststore_password" : "$conf{truststore.password}",
+      "identity.auth_framework.endpoint.identity_server_service_url" : "https://localhost:9443/services/",
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/infer.json]: Host and port should use the placeholders

File: modules/distribution/src/repository/resources/conf/infer.json [hunk 2]
```diff
@@ -115,5 +115,37 @@
...
+      "identity.auth_framework.endpoint.username_header" : "UserName",
+      "identity.auth_framework.endpoint.key_manager_type" : "SunX509",
+      "identity.auth_framework.endpoint.trust_manager_type" : "SunX509",
+      "identity.auth_framework.endpoint.tls_protocol" : "TLSv1.2"
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/infer.json]: These may also have placeholder references. please check

File: modules/distribution/src/repository/resources/conf/infer.json [hunk 3]
```diff
@@ -115,5 +115,37 @@
...
+      "identity.auth_framework.endpoint.hostname_verification_enabled" : true,
+      "identity.auth_framework.endpoint.mutual_ssl_username" : "admin",
+      "identity.auth_framework.endpoint.client_keystore" : "$conf{keystore.tls.file_name}",
+      "identity.auth_framework.endpoint.carbon_security_keystore_password" : "$conf{keystore.tls.password}",
+      "identity.auth_framework.endpoint.client_truststore" : "$conf{truststore.file_name}",
+      "identity.auth_framework.endpoint.carbon_security_truststore_password" : "$conf{truststore.password}",
+      "identity.auth_framework.endpoint.identity_server_service_url" : "https://localhost:9443/services/",
+      "identity.auth_framework.endpoint.username_header" : "UserName",
+      "identity.auth_framework.endpoint.key_manager_type" : "SunX509",
+      "identity.auth_framework.endpoint.trust_manager_type" : "SunX509",
```
Comment (pulasthi7) [inline modules/distribution/src/repository/resources/conf/infer.json]: do the same as commented above for these also

File: modules/distribution/src/repository/resources/conf/infer.json (hunk 2 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/infer.json]: Fixed.

File: modules/distribution/src/repository/resources/conf/infer.json (hunk 1 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/infer.json]: Fixed.

File: modules/distribution/src/repository/resources/conf/infer.json (hunk 3 above)
Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/infer.json]: Fixed.

File: modules/distribution/src/repository/resources/conf/default.json [hunk 4]
```diff
@@ -186,5 +186,9 @@
       "type": "httpbasic",
       "primary": false
     }
-  }
+  },
+
```
Comment (GDRDABARERA) [inline modules/distribution/src/repository/resources/conf/default.json]: Unnecessary new line

Comment (hasinidilanka) [reply modules/distribution/src/repository/resources/conf/default.json]: These files were moved to the component level and refactored.

**Pull request merged**